template/pkg/command/server.go

```diff
@@ -134,17 +137,18 @@ func Server(cfg *config.Config) *cli.Command {
 			{
 				server, err := http.Server(
 					http.Logger(logger),
+					http.Namespace(httpNamespace),
 					http.Context(ctx),
 					http.Config(cfg),
 					http.Metrics(metrics),
-					http.Flags(flagset.RootWithConfig(cfg)),
-					http.Flags(flagset.ServerWithConfig(cfg)),
+					http.Flags(flagset.RootWithConfig(config.New())),
```

Comment:
Not sure if this is right, it will pass an empty config instance to every config part thus loosing the values for previous settings.

Reply:
Instantiate an empty config struct once and pass it trough all the functions with &

Reply:
@refs This is from ocis-proxy

Reply:
ocis-phoenix does the same.
ocis-accounts not

Reply:
Ah okay, I understand now, the http.Flags() method does the merging.

Reply:
This is because of go-micro, deep beneath the surface when the command runs it does validate all the flags the command was called with, and because we are wrapping go-micro cli with our cli (which is the same library) we need to make sure that go-micro is aware of our flags. The problem comes in that our `flagset` set of methods, as @IljaN said, merges variables, and since we toss a `*config.Config` (a pointer) around, this values are override, and since the library only care about the flag definition, an empty set of flags is passed, so values are not override.